versioned/storage/common/src/main/java/org/projectnessie/versioned/storage/common/logic/CommitLogicImpl.java

```diff
@@ -599,7 +607,7 @@ public CommitObj buildCommitObj(
       StoreIndexElement<CommitOp> existing = existingFromIndex(fullIndex, key);
       CommitOp existingContent = existing != null ? existing.content() : null;
 
-      ObjId expectedValue = remove.expectedValue();
+      ObjId expectedValue = expectedValueReplacement.maybeReplaceValue(key, remove.expectedValue());
```

Comment:
Not sure, I follow, sorry :thinking: The `CreateCommit` parameter is "pure" input, it does not contain any data from previous commits, it can be constructed without knowing previous content on the target branch.

Reply:
I tried it - it would need to to lots of things that buildCommit() already does - so it'd end up w/ a lot of duplicate work.

Reply:
noted - thx :+1: